### ISSUE TITLE
Gas Pricing Strategy

### ISSUE THREAD

**Issue by AdityaSripal** [cosmos/ethermint-archive#473]
Decide on a gas pricing strategy for Ethermint (i.e. how much gas does each operation cost). There are two obvious choices:

1. Use SDK's pricing strategy
2. Use Ethereum's pricing strategy

I think it makes sense to use Ethereum's pricing. We will probably use that anyway for the gas cost of each opcode during a smart contract call, so it makes sense to use it everywhere else as well.

**Comment by AdityaSripal** [comment]
cc @alexanderbez

**Comment by mslipper** [comment]
Mind looping me in on the discussion as well? I'd like whatever we implement to be exposed via the usual Web3 RPC methods.

**Comment by alexanderbez** [comment]
For sure @mslipper

**Comment by ValarDragon** [comment]
~This is our chance to fix Etheruem's gas pricing! This is a known problem with ethereum. I think we should look at all the problems with Ethereum's existing gas pricing and incorporate fixes for them.~

Upon further thought, I think we should use Ethereums existing gas pricing, so all existing tooling and contracts "just work", and its not an annoying edge case thats a barrier to adoption. We can make a better pricing scheme after adoption (or make an easy way to alter the pricings)

**Comment by alexanderbez** [comment]
@ValarDragon yes, would love as much as your input as possible. But you hit the nail on the head:
> Upon further thought, I think we should use Ethereums existing gas pricing, so all existing tooling and contracts "just work".

This is our ultimate aim at release.
